lib/gemstash/cli.rb

```diff
@@ -66,5 +67,20 @@ def status
     def stop
       Gemstash::CLI::Stop.new(self).run
     end
+
+    desc "preload", "Preloads all the gems in your gemstash server from the default upstream"
+    method_option :latest, :type => :boolean, :default => false, :desc =>
+      "Only fetch the latest specs"
+    method_option :server_url, :type => :string, :default => "http://localhost:9292", :desc =>
+      "Gemstash server url"
+    method_option :threads, :type => :numeric, :default => 20, :desc =>
+      "Number of threads to run the fetching job with"
+    method_option :limit, :type => :numeric, :default => nil, :desc =>
+      "Limit for the number of gems to preload"
+    method_option :skip, :type => :numeric, :default => 0, :desc =>
+      "Number of gems to skip when preloading"
+    def preload
```

Comment:
Yeah, it is tricky.

What I like about the client approach is that we can exercise the API and not have special cases for things like batch import or this other batch preload case. That simplifies the server as we do not have to branch the code depending where the call comes from.

On the other hand, it does have specific needs (like celluloid if we ever want to us it), but I do not see much more needs added, at least in this particular case, to justify extracting a gem to not affect gemstash with client specific dependencies.

What do you think about this @indirect ?